1.0.0 doc changes

cs-abhishek-shukla opened the pull request:
- Added screenshots in usage
- Elaborated README.md with possible use cases
- Added an SS with custom data in setup.md
- Fixed typos

cs-abhishek-shukla committed:
1.0.0 doc changes
- Added screenshots in usage
- Elaborated README.md with possible use cases
- Added an SS with custom data in setup.md
- Fixed typos

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -10,23 +10,25 @@
+| Fields                | Description                                                                         |
+|-----------------------|-------------------------------------------------------------------------------------|
+| Title                 | Specify a title for the SOC Overview Sankey widget as it appears on the dashboard.  |
+| Resource              | Select the module whose Sankey representation is to be displayed.                   |
```
suggestion: whose Sankey representation is to be displayed, i.e. FortiSOAR module

cs-abhishek-shukla replied:
Fixed.

cs-suyog-jinturkar commented on README.md:
```diff
@@ -12,7 +12,33 @@
+8. **Real-time Monitoring:**
+   - Integrate the Sankey chart into a real-time monitoring dashboard within FortiSOAR to provide FortiSOAR&trade; analysts with a dynamic and up-to-date view of the security operations flow.
+
+When implementing a Sankey chart widget in FortiSOAR, ensure that it supports dynamic updates and real-time data feeds to provide a current and accurate representation of the FortiSOAR&trade; workflow. Additionally, customize the chart to meet the specific needs and processes of your FortiSOAR&trade; environment.
```
ensure that it supports dynamic updates and real-time data feeds to provide a current and accurate representation of the FortiSOAR&trade; workflow ---> What is expected by the user here?

cs-suyog-jinturkar replied:
We might need to add more details or example here for the user

cs-abhishek-shukla replied:
Removed the line... the updates are real-time by default

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -10,23 +10,25 @@
+| Title                 | Specify a title for the SOC Overview Sankey widget as it appears on the dashboard.  |
+| Resource              | Select the module whose Sankey representation is to be displayed.                   |
+| Filter Criteria       | Define the filter criteria using which to filter the data retrieved by this widget. |
+| Source Nodes          | Select the field to render source nodes.                                            |
```
any field or picklist field?

cs-abhishek-shukla replied:
Fixed.

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -10,23 +10,25 @@
+Here JSON object requires 2 array properties, **`Nodes`** and **`Links`**. Value in **`Nodes`** act as a *`Source`* node for the module selected in the **Resource** field.
+
+Following is a sample JSON that can be entered in the **External Source JSON** field.
+
```
as per me, Data mentioned in the json is not same as data in the screenshot. Please correct, if needed.

cs-suyog-jinturkar commented on docs/setup.md:
````diff
@@ -10,23 +10,25 @@
+
+Following is a sample JSON that can be entered in the **External Source JSON** field.
+
 ```JSON
````
@cs-vikas-shelar please add more details around input json and its usage here.

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,51 +3,82 @@
-- Visualize relationship data as per duration. User can select from standard durations provided as part of widget like 7 Days, 1 Day, and 1 Hour. 
+You can select from following options:
 
+- 7 Days
```
last 7 days, last 1 day, etc.

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,51 +3,82 @@
+
+    ![](./res/sankey-edit-04.png)
 
 6. Select the text field to be grouped for source nodes, in the **Source Nodes** field.
```
text field or picklist?

cs-abhishek-shukla replied:
fields of type text listed in the picklist

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,51 +3,82 @@
+
+    ![](./res/sankey-edit-06.png)
 
 8. Select the relationship module, to group data to visualize as per relation, in the **Relationship Resource** field.
```
We should describe this point better by giving more details around, what is relationship module means here?

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,51 +3,82 @@
+
 ### SOC Overview Sankey Widget - View
 
 The following image displays a **SOC Overview Sankey** widget based on an example where you might want to view the data as per severity of related incidents of alerts group by source and severity:
```
We should also render and illustrate one more example here taken from your overview section, say alert --> type --> severity --> escalated/not escalated or you can pick other example as well here

cs-suyog-jinturkar reviewed (commented):
Added review comments

cs-vikas-shelar commented on docs/setup.md:
```diff
@@ -10,47 +10,149 @@
+| Filter Criteria       | Define the filter criteria using which to filter the data retrieved by this widget.                                                            |
+| Source Nodes          | Select the a field, within the **_Resource_** module, to render source nodes. Only the fields of type **_text_** are listed.                   |
+| Target Nodes          | Select the picklist, within the **_Resource_** module, to render target nodes. Only the fields of type **_picklist_** are listed.              |
+| Relationship Resource | Select the FortiSOAR&trade; module, which has a relationship to the module selected in the **Source Node** field. For example, **_Incidents_** |
```
to the module selected in the **Resource** field.

cs-vikas-shelar reviewed (commented):
The changes look good to me. 1 minor change is required for field detail.

cs-abhishek-shukla committed:
Incorporated feedback
- The custom json contains data to generate the Sankey chart shown in the doc
- Fixed setup section to better describe the fields
- Updated screenshots
- Detailed external source json

cs-vikas-shelar reviewed (approved):
Looks good to me.
